### PR TITLE
[FIX][QAT] Switch to use `kwargs` when `args` is empty

### PR DESCRIPTION
**Pull request by sijiac**
Summary:
When `ref_node.args` is empty, the QAT will throw index out of range. Here is an example, line 574 is using `tensors = ....` in torch.cat func, which will be treated as `kwargs`
{F800357376}

f388506954

To fix the issue, we will use the value of the first kwarg if args is empty

Test Plan: f388545532

Reviewed By: bigning, lyoka

Differential Revision: D41396771

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

### CODE REVIEW

**Review by lyoka — APPROVED** [review]

**Review by houseroad — APPROVED** [review]
Looks good.

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -829,7 +829,8 @@ def special_pattern_replacement(model: QuantizedGraphModule):
         is_call_function, is_call_method, is_call_module = is_special_pattern_node(ref_node, modules)
         if not (is_call_module or is_call_function or is_call_method):
             continue
-        dq_node_or_nodes = ref_node.args[0]
+        assert len(ref_node.args) > 0 or len(ref_node.kwargs) > 0
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:832]: if it works for your case, maybe we can start with requiring len(ref_node.kwargs) == 1

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 2]
```diff
@@ -829,7 +829,8 @@ def special_pattern_replacement(model: QuantizedGraphModule):
...
+        dq_node_or_nodes = ref_node.args[0] if len(ref_node.args) > 0 else list(ref_node.kwargs.items())[0][1]
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:833]: nit: `list(ref_node.kwargs.items())[0][1]` --> `list(ref_node.kwargs.values())[0]`

File: torch/ao/quantization/fx/_lower_to_native_backend.py (hunk 1 above)
Comment (sijiac) [reply torch/ao/quantization/fx/_lower_to_native_backend.py:832]: for `torch.cat`, there would be two kwargs, `tensors` and `dim`. It may not work if we limit the number to be 1
